apps/src/aichat/views/bot/MovingBot.tsx

```diff
@@ -0,0 +1,70 @@
+import React, {useState} from 'react';
+import Bot, {BotState} from './Bot';
+
+function pairwise(arr: [], func: Function) {
+  for (var i = 0; i < arr.length - 1; i++) {
+    func(arr[i], arr[i + 1]);
+  }
+}
+
+interface MovingBotProps {
+  currentTick: number;
+  moves: any;
+}
+
+function easeInExpo(t: number, b: number, c: number, d: number) {
+  return t == 0 ? b : c * Math.pow(2, 10 * (t / d - 1)) + b;
+}
+
+const MovingBot: React.FunctionComponent<MovingBotProps> = ({
+  currentTick,
+  moves,
+}) => {
+  let botState = {
+    head: [110, 10],
+    body: [140, 150],
+    hand0: [25, 130],
+    hand1: [250, 130],
+    foot0: [60, 290],
+    foot1: [210, 290],
+  };
+
+  // Convert moves into current coordinates of each piece, based on current time.
+
+  for (const partKey of Object.keys(moves)) {
+    pairwise(moves[partKey], (move: any, nextMove: any) => {
```

Comment:
If the `pairwise` function and `moves` prop have the correct types then you can delete both `any`s here and the types will be inferred as `MoveBlockStep`.